pkg/sql/parser/datum.go

```diff
@@ -2001,14 +2001,22 @@ func (*DArray) AmbiguousFormat() bool { return false }
 
 // Format implements the NodeFormatter interface.
 func (d *DArray) Format(buf *bytes.Buffer, f FmtFlags) {
-	buf.WriteByte('{')
+	if f.disambiguateDatumTypes {
+		buf.WriteString("ARRAY[")
```

Comment:
As I said on the other issue I would just go for the `ARRAY[...]` notation all the time. The code would become simpler.